### ISSUE TITLE
WinRM error when running inspec verifier on windows 2008 with kitchen test

### ISSUE THREAD

**Issue by mhedgpeth** [inspec/inspec#784]
### Description

When I run `kitchen test` I get a winrm error. When I do a `kitchen converge` afterwards, the tests pass.
### InSpec and Platform Version

0.24, kitchen 1.8
### Replication Case

Use `ncr/win2008r2` as your virtual box for vagrant. Run `kitchen test` with these tests:

```
control 'd-drive-should-exist' do
  impact 1
  title 'Image should have a D: on it'
  desc 'The images used by kitchen should have a D: on them to make them like Production'
  tag 'chef-312'
  describe directory('D:/') do
    it { should exist }
  end
end

control 'chef-client-should-be-available' do
  impact 1
  title 'Chef Client should be installed'
  desc 'To save time with kitchen converges, chef-client should be installed'
  describe command('chef-client') do
    it { should exist }
  end
end
```

You will get an error when you try to run the tests:

```
  1) Command chef-client should exist
     Failure/Error: fail WinRMWSManFault.new(fault.to_s, fault.attributes['Code']) unless fault.nil?

     WinRM::WinRMWSManFault:
       [WSMAN ERROR CODE: 2150859174]: <f:WSManFault Code='2150859174' Machine='127.0.0.1' xmlns:f='http://schemas.micro
soft.com/wbem/wsman/1/wsmanfault'><f:Message>The WS-Management service cannot process the request. This user is allowed
a maximum number of 15 concurrent operations, which has been exceeded. Close existing operations for this user, or raise
 the quota for this user. </f:Message></f:WSManFault>
     # C:/Users/mh185122/AppData/Local/chefdk/gem/ruby/2.1.0/gems/inspec-0.24.0/lib/resources/os.rb:33:in `[]'
     # C:/Users/mh185122/AppData/Local/chefdk/gem/ruby/2.1.0/gems/inspec-0.24.0/lib/resources/command.rb:48:in `exist?'
     # ./test/integration/default/inspec/image.rb:17:in `block (3 levels) in load'
     # C:/Users/mh185122/AppData/Local/chefdk/gem/ruby/2.1.0/gems/inspec-0.24.0/lib/inspec/runner_rspec.rb:66:in `run'
```
### Possible Solutions

This could not be related to inspec but could be related to winrm in kitchen. I'll post this internally with Chef on the windows channel as well.

**Comment by mwrock** [comment]
It may be helpful to get a debug log and grab a better stack trace. The winrm gem has logic to avoid this scenario. 2k8r2 has a very low command threshold and it counts the commands it issues and reset the connection before hitting the 15 command limit.

So far this has not blown up but the whole strategy of command counting can be precarious and winrm v2 will take a different approach and respond to this error by reseting the connection if the error occurs. So there must be some command not getting counted in this scenario and causing this to fail.

**Comment by mhedgpeth** [comment]
@mwrock are you saying to run kitchen in debug mode & post the logs from `.kitchen/logs`? Or should I get logging from the image itself?

**Comment by mwrock** [comment]
yeah. apologies for the VERY roundabout way of communicating that.

**Comment by mhedgpeth** [comment]
Not a problem, here you go, @mwrock 
[kitchen.zip](https://github.com/chef/inspec/files/302716/kitchen.zip)

**Comment by mwrock** [comment]
just took a look at those logs and they don't appear to have the above error. Is it possible you attached the wrong zip?

**Comment by mhedgpeth** [comment]
@mwrock sorry for the delay on this. I reran it not in debug mode and it failed again. Then I ran it in debug mode and it passed. So perhaps the debugging slows it down to get rid of more connections as they time out?

Also I saw that you added more logging to the next version of test kitchen. Might that make it easier to figure out?

**Comment by mwrock** [comment]
I don't think that logging will help here unfortunately. When v2 comes out and gets integrated down stream, that should definitely eliminate this bug. Thats not to say we cant push a fix ihere if it vecomes clear.

**Comment by mhedgpeth** [comment]
Thanks @mwrock. What's the timeline on v2?

**Comment by mwrock** [comment]
I'm hoping to wrap it up in the next couple weeks. I've instructed myself not to do any significant side OSS work on other projects until this is done.

**Comment by chris-rock** [comment]
@mhedgpeth is this still an issue?

**Comment by mwrock** [comment]
FYI: just waiting on winrm-v2 review - https://github.com/WinRb/WinRM/pull/191 now. It definitely addresses this issue.

**Comment by mhedgpeth** [comment]
Thanks @mwrock I'm watching that issue. I can retest when it's merged.

**Comment by jerryaldrichiii** [comment]
@mhedgpeth, is this still an issue?

**Comment by kekaichinose** [comment]
__What’s happening? Why was this issue closed?__
This issue was closed due to some much needed review of legacy issues or issues that were spawned in older versions of InSpec, i.e. < v3.

__Why do I care?__
You would care about this if this was an issue that you are still seeing and/or feel needs to be addressed in the current version of InSpec.

__What do I need to do?__
If this issue is no longer important, no further action is necessary. However, if you think this is something that should be addressed, please open a new issue and refer to the original issue in the description.
